app/main.py

```diff
@@ -2,4 +2,54 @@ class Cargo:
     def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot():
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords or [0, 0]
+
+    def go(self, axis: int, direction: int, steps: int = 1) -> None:
+        self.coords[axis] += direction * steps
+
+    def go_right(self, steps: int = 1) -> None:
+        self.go(0, 1, steps)
+
+    def go_left(self, steps: int = 1) -> None:
+        self.go(0, -1, steps)
+
+    def go_forward(self, steps: int = 1) -> None:
+        self.go(1, 1, steps)
+
+    def go_back(self, steps: int = 1) -> None:
+        self.go(1, -1, steps)
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        super().__init__(name, weight)
+        self.coords = coords or [0, 0, 0]
+
+    def go_up(self, steps: int = 1) -> None:
+        self.go(2, 1, steps)
+
+    def go_down(self, steps: int = 1) -> None:
+        self.go(2, -1, steps)
+
+
+class DeliveryDrone(FlyingRobot):
+    def __init__(self, name: str, weight: int, max_load_weight: int,
+                 coords: list = None, current_load: int = None) -> None:
```

Comment:
use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contains string elements and can be None has typehint Optional[list[str]]

add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]
```suggestion
    def __init__(
                 self,
                 name: str,
                 weight: int,
                 max_load_weight: int,
                 coords: list = None,
                 current_load: int = None
                 ) -> None:
```